### PR TITLE
Refactor MTE-3854 WIP: Menu Refactor Cherry Pick (Smoke Test only)

### PR DESCRIPTION
**Pull request by clarmso**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-TODO)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/TODO)

## :pencil: Checklist
You have to check all boxes before merging
- [ ] Filled in the above information (tickets numbers and description of your work)
- [ ] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: firefox-ios/firefox-ios-tests/Tests/UnitTest.xctestplan [hunk 1]
```diff
@@ -111,6 +111,7 @@
         "HistoryHighlightsDataAdaptorTests\/testReloadDataOnNotification()",
         "HistoryHighlightsTests",
         "IntroViewControllerTests\/testBasicSetupReturnsExpectedItems()",
+        "PocketViewModelTests\/testClickingDiscoverCell_recordsTapOnStory()",
```
Comment (clarmso) [inline firefox-ios/firefox-ios-tests/Tests/UnitTest.xctestplan:114]: No idea why those unit tests fail in Bitrise. :(

File: firefox-ios/firefox-ios-tests/Tests/XCUITests/BookmarksTests.swift [hunk 2]
```diff
@@ -392,7 +383,7 @@ class BookmarksTests: BaseTestCase {
 
     private func bookmarkPageAndTapEdit() {
         bookmark()
-        app.buttons["Edit"].waitAndTap()
+        bookmark()
```
Comment (isabelrios) [inline firefox-ios/firefox-ios-tests/Tests/XCUITests/BookmarksTests.swift:386]: do we need to go to bookmark() twice?

**Comment by isabelrios** [comment]
@clarmso how is this work going? I just leave a question, apart from that, when do you expect this could land?

File: firefox-ios/firefox-ios-tests/Tests/XCUITests/BookmarksTests.swift (hunk 2 above)
Comment (clarmso) [reply firefox-ios/firefox-ios-tests/Tests/XCUITests/BookmarksTests.swift:386]: This private function is for `testEditBookmark()`, a full functional test. This test bookmarks a page first and then opens the Bookmark page. Let me take a look what's going on here.

**Comment by clarmso** [comment]
> @clarmso how is this work going? I just leave a question, apart from that, when do you expect this could land?

This PR can land when the menu refactor is enabled. The `bookmark()` related question is for a full functional tests.

We will address the issue on full functional tests after this PR is landed so that the developers are not blocked.

File: firefox-ios/firefox-ios-tests/Tests/XCUITests/BookmarksTests.swift (hunk 2 above)
Comment (clarmso) [reply firefox-ios/firefox-ios-tests/Tests/XCUITests/BookmarksTests.swift:386]: Yes, we need to call `bookmark()` twice. Before the menu refactor, there's an "Edit" button from the toast to access the bookmarks panel right away.
Simulator Screenshot - iPhone 15 Pro - 2024-11-21 at 16 41 44

After the menu refactor, there's no such a button. The way to open the bookmarks panel is to open the browser tab menu ➡️ Save ➡️ Edit Bookmarks (was "Add Bookmarks"). The second `bookmarks()` call opens the bookmarks panel.
Simulator Screenshot - iPhone 15 - 2024-11-21 at 17 20 39

File: firefox-ios/firefox-ios-tests/Tests/UnitTest.xctestplan [hunk 3]
```diff
@@ -119,6 +120,7 @@
         "ShortcutRouteTests\/testOpenLastBookmarkShortcutWithInvalidUrl()",
         "TabManagerTests\/testDeleteSelectedTab()",
         "TabManagerTests\/testPrivatePreference_togglePBMDeletesPrivate()",
+        "TabToolbarHelperTests\/testTelemetryForSiteMenu()",
```
Comment (clarmso) [inline firefox-ios/firefox-ios-tests/Tests/UnitTest.xctestplan:123]: We can disable these unit test temporarily.

**Comment by clarmso** [comment]
FYI: I have not tested the changes with Xcode 16 and iOS 18 yet.

**Comment by isabelrios** [comment]
Can you rebase with latest main to see the changes working on xcode 16? the experiment will be enabled within this PR so that the change and tests land at the same time as we did with toolbar .. 
cc @thatswinnie

**Comment by clarmso** [comment]
Sorry folks. Rebase failed again. Let me cherry pick the smoke tests-related changes momentarily. 😢

**Comment by clarmso** [comment]
Clean PR here: https://github.com/mozilla-mobile/firefox-ios/pull/23381

**Pull request closed without merging**